cli/lightning-cli.c

```diff
@@ -282,10 +311,10 @@ int main(int argc, char *argv[])
 		}
 		tal_append_fmt(&cmd, "} }");
 	} else {
-		tal_append_fmt(&cmd, "[ ");
-		for (i = 2; i < argc; i++)
-			add_input(&cmd, argv[i], i, argc);
-		tal_append_fmt(&cmd, "] }");
+		if (streq(method, "check"))
+			default_check_input(&cmd, argv, argc);
+		else
+			default_input(&cmd, argv, argc);
```

Comment:
This is a bit awkward, and doesn't allow the keyword case.

Perhaps we're better off doing this work in the server; hoist 'cmd' out of params manually there?  The main reason for that is because there are other clients, like the python library.

Reply:
Agreed!  This was the easiest way to get the feature in and see if you  liked it.  I'll get to work on manually constructing the command in the check handler next.